backend/services/AccountService.ts

```diff
@@ -0,0 +1,219 @@
+    if (!AccountService.instance) {
+      AccountService.instance = new AccountService();
+      await AccountService.instance.connect();
+    }
+    return AccountService.instance;
+  }
+
+  private async connect() {
+    if (this._client === undefined) {
+      const api = new OpenAPIClientAxios({
+        definition: openapiJson as Document,
+        withServer: { url: Config.instance().accountServiceUrl },
+      });
+      this._client = await api.init<AccountServiceClient>();
+    }
+  }
+
+  private set client(api: AccountServiceClient) {
+    this._client = api;
+  }
+
+  private get client() {
+    if (this._client === undefined) {
+      throw new Error(`${this.constructor.name} API not initialized`);
+    }
+    return this._client;
+  }
+
+  /**
+   * Retrieves the SIWF configuration for wallet login.
+   * @returns A promise that resolves to a `WalletLoginConfigResponse` object.
+   * @throws If there was an error retrieving the SWIF config.
+   */
+  public async getSWIFConfig(): Promise<WalletLoginConfigResponse> {
+    try {
+      const response = await this.client.AccountsController_getSIWFConfig();
+      return response.data;
+    } catch (e) {
+      logger.error('Failed to get SIWF config: ', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Retrieves the account information for a given MSA ID.
+   * @param msaId - The MSA ID of the account to retrieve.
+   * @returns A Promise that resolves to an AccountResponse object containing the account information.
+   * @throws If there was an error retrieving the account information.
+   */
+  public async getAccount(msaId: string): Promise<AccountResponse> {
+    try {
+      const response = await this.client.AccountsController_getAccount(msaId);
+      logger.debug(
+        `AccountService: getAccount: Got account for msaID:(${msaId}), data:(${JSON.stringify(response.data)})`
+      );
+      return {
+        msaId: parseInt(msaId),
+        displayHandle: response.data.displayHandle,
+      };
+    } catch (e) {
+      logger.error(`Failed to get account for msaID:(${msaId}) error:${e}`);
+      throw e;
+    }
+  }
+
+  /**
+   * Retrieves an account based on the provided reference ID.
+   * @param referenceId - The reference ID used to correlate the data from the blockchain transaction to the account.
+   * @returns A Promise that resolves to an AccountResponse object if the account is found, or undefined if not found.
+   * @throws Throws an error if there was an issue retrieving the account.
+   */
+  public async getAccountByReferenceId(referenceId: string): Promise<AccountResponse | undefined> {
+    // In this case, we're using the referenceId to get the account
+    // Check the webhook and see if the referenceId has been processed
+    logger.debug(`AccountService: getAccountByReferenceId: Looking for account for referenceId:(${referenceId})`);
+    try {
+      const accountData = WebhookController.referenceIdsReceived.get(referenceId);
```

Comment:
I think maybe move `referenceIdsReceived` out of the controller class and into a service; seems like it would be more consistent with the design. `WebhookController` is meant to be super-lightweight, as it's serving webhooks across all domains.